device_backends/include/NumericAddressedLowLevelTransferElement.h

```diff
@@ -34,24 +34,24 @@ namespace ChimeraTK {
   class NumericAddressedLowLevelTransferElement : public TransferElement {
    public:
     NumericAddressedLowLevelTransferElement(
-        boost::shared_ptr<NumericAddressedBackend> dev, size_t bar, size_t startAddress, size_t numberOfWords)
+        boost::shared_ptr<NumericAddressedBackend> dev, size_t bar, size_t startAddress, size_t numberOfBytes)
     : TransferElement("", {AccessMode::raw}), _dev(dev), _bar(bar), isShared(false) {
       if(bar > 5 && bar != 13) {
         std::stringstream errorMessage;
         errorMessage << "Invalid bar number: " << bar << std::endl;
         throw ChimeraTK::logic_error(errorMessage.str());
       }
-      setAddress(startAddress, numberOfWords);
+      setAddress(startAddress, numberOfBytes);
     }
 
     virtual ~NumericAddressedLowLevelTransferElement() {}
 
     void doReadTransferSynchronously() override {
-      _dev->read(_bar, _startAddress, rawDataBuffer.data(), _numberOfBytes);
+      _dev->read(_bar, _startAddress, (int32_t*)rawDataBuffer.data(), _numberOfBytes);
```

Comment:
Hm, strictly speaking I should also cast via void* because of the strict aliasing rule (which in this context does not matter because there is no surrounding code and it is a virtual function, so the optimiser cannot do anything wrong here - but one should probably not rely on that). So it will become:
 `reinterpret_cast<int32_t*>(reinterpret_cast<void*>(rawDataBuffer.data()))`.

This is why I sometimes actually prefer C style casts, because:
`(int32_t*)(void*)rawDataBuffer.data()`
is actually more readable and does clearly the same thing.

Does our coding style rule generally forbid C style casts?

Reply:
done